File: packages/flutter_background_service_android/android/src/main/java/id/flutter/flutter_background_service/NotificationData.java
package id.flutter.flutter_background_service;

public final class NotificationData {
    private final String day;
    private final String jalali;
    private final String miladi;
    private final String hijri;
    private final String theme;

    public NotificationData(String day, String jalali, String miladi, String hijri, String theme) {
        this.day = day;
        this.jalali = jalali;
        this.miladi = miladi;
        this.hijri = hijri;
        this.theme = theme;
    }

    public String getDay() {
        return day;
    }

    public String getJalali() {
        return jalali;
    }

    public String getMiladi() {
        return miladi;
    }

    public String getHijri() {
        return hijri;
    }

    public String getTheme() {
        return theme;
    }
}

File: packages/flutter_background_service_platform_interface/lib/src/notification_data.dart
import 'dart:convert' show jsonEncode;

class NotificationData {
  final String day;
  final String jalali;
  final String miladi;
  final String hijri;
  final String theme;
  const NotificationData({
    required this.day,
    required this.jalali,
    required this.miladi,
    required this.hijri,
    required this.theme,
  });
  Map<String, dynamic> _toMap() => {'day': day, 'jalali': jalali, 'miladi': miladi, 'hijri': hijri, 'theme': theme};
  String toJson() => jsonEncode(_toMap());
}
